skyllh/core/dataset.py

```diff
@@ -2445,6 +2445,31 @@ def __str__(self):
 
         return lines
 
+    def add_aux_data(
+            self,
+            name,
+            data,
+    ):
+        """Adds the given data as auxiliary data to all datasets of this
+        dataset collection.
+
+        Parameters
+        ----------
+        name : str
+            The name under which the auxiliary data will be stored.
+        data : unspecified
+            The data that should get stored. This can be of any data type.
+
+        Raises
+        ------
+        KeyError
+            If auxiliary data is already stored under the given name.
+        """
+        for dataset in self._datasets.values():
```

Comment:
I think this function should return an error if called before any Dataset object has been added to the dataset collection.

Reply:
I fixed that.